src/wasm-linker.h

```diff
@@ -32,6 +32,9 @@ namespace wasm {
 
 class S2WasmBuilder;
 
+// Name of the dummy function to prevent erroneous nullptr comparisons.
+static constexpr const char* dummyFunction = "__wasm_nullptr";
```

Comment:
wasm-emscripten.cpp just needs to know that it shouldn't generate thunks for that, right? In other words, a dyncall thunk is just an indirect way of exporting a function. We could add some sort of interface to the linker (or somewhere else) saying "can this function be exported". Currently the dummy function and functions taking i64 are excluded. Unfortunately I'd consider the i64 exclusion to be more a property of the emscripten/web embedding than the linker proper though, so maybe that shouldn't go in the linker's version of what to exclude. Even so it might still be cleaner to have some sort of `Linker::canExport(Name function)` method than having the name itself be shared.